lib/oxigraph/src/storage/error.rs

```diff
@@ -56,59 +32,42 @@ impl From<StorageError> for io::Error {
 }
 
 /// An error return if some content in the database is corrupted.
-#[derive(Debug)]
-pub struct CorruptionError {
-    inner: CorruptionErrorKind,
-}
+#[derive(Debug, thiserror::Error)]
+#[error(transparent)]
+pub struct CorruptionError(#[from] CorruptionErrorKind);
 
-#[derive(Debug)]
+/// An error return if some content in the database is corrupted.
+#[derive(Debug, thiserror::Error)]
 enum CorruptionErrorKind {
+    #[error("{0}")]
     Msg(String),
-    Other(Box<dyn Error + Send + Sync + 'static>),
+    #[error("{0}")]
+    Other(#[source] Box<dyn Error + Send + Sync + 'static>),
 }
 
 impl CorruptionError {
     /// Builds an error from a printable error message.
     #[inline]
     pub(crate) fn new(error: impl Into<Box<dyn Error + Send + Sync + 'static>>) -> Self {
-        Self {
-            inner: CorruptionErrorKind::Other(error.into()),
-        }
+        Self(CorruptionErrorKind::Other(error.into()))
     }
 
-    /// Builds an error from a printable error message.
     #[inline]
-    pub(crate) fn msg(msg: impl Into<String>) -> Self {
-        Self {
-            inner: CorruptionErrorKind::Msg(msg.into()),
-        }
+    pub(crate) fn from_encoded_term(encoded: &EncodedTerm, term: &TermRef<'_>) -> Self {
+        // TODO: eventually use a dedicated error enum value
+        Self::new(format!("Invalid term encoding {encoded:?} for {term}"))
```

Comment:
Nit: Is is building a `CorruptionErrorKind::Other(Box<dyn String>)`. What about using `Self::msg` to get a `CorruptionErrorKind::Msg(String)` instead?

Reply:
Sure - I was trying to keep original error codes - you used this before:

```
CorruptionError::new(format!("Invalid term encoding {encoded:?} for {term}")).into())
```

fixed

Reply:
It seems I got it wrong. sorry for that